doc/debugger.rst

```diff
@@ -922,6 +935,8 @@ Error codes
 +--------+------------------------------------------------------------------+
 | 0x03   | Not found (e.g. invalid breakpoint index)                        |
 +--------+------------------------------------------------------------------+
+| 0x04   | AppRequest error (i.e. request callback returned -1)             |
```

Comment:
A small addition to this: I was thinking that maybe adding an "Application error" as a generic type might be useful to distinguish, as you say, errors originating from the application vs. Duktape.

At first I thought this is already apparent because the command is AppRequest. But even in this case it's possible for Duktape to return an error (say "application doesn't support AppRequest") which is technically different from an application returned error (say "failed to reset device: i/o device permission error"). They still probably don't have specific error handling in the debug client so maybe it doesn't matter.

So maybe if we add an error it could be "Application error"? Does that make sense to you?